core/inputs.c

```diff
@@ -442,6 +434,21 @@ void postTouchEvent(uint8_t ID, float x, float y, float dx, float dy, TouchState
     te = NULL;
 }
 
+Touch* getDownTouches(int* nbTouches) {
+    InputContext *c = inputContext();
+
+    Touch t;
+    Touch out[NB_TOUCHES];
+    for (int id = 0; id < NB_TOUCHES; id +=1) {
+        t = c->touches[id];
+        if (t.down || t.state == TouchStateDown) {
+            *nbTouches += 1;
+            out[id] = t;
+        }
+    }
+    return out;
```

Comment:
CI error here.

you cannot return the address of a local variable.
You need to use malloc to create `out` variable.

```cpp
Touch *out = malloc(sizeof(Touch) * NB_TOUCHES);
```